salt/modules/telemetry.py

```diff
@@ -0,0 +1,348 @@
+        is specified, obtain the specific alert config
+
+    Returns dictionary or list of dictionaries.
+
+    CLI Example:
+
+        salt myminion telemetry.get_alert_config rs-ds033197 currentConnections profile=telemetry
+        salt myminion telemetry.get_alert_config rs-ds033197 profile=telemetry
+    """
+
+    auth = _auth(profile=profile)
+    alert = False
+
+    key = "telemetry.{0}.alerts".format(deployment_id)
+
+    if key not in __context__:
+        try:
+            get_url = _get_telemetry_base(profile) + "/alerts?deployment={0}".format(deployment_id)
+            response = requests.get(get_url, headers=auth)
+        except requests.exceptions.RequestException as e:
+            log.error(str(e))
+            return False
+
+        http_result = {}
+        if response.status_code == 200:
+            for alert in response.json():
+                http_result[alert.get('condition', {}).get('metric')] = alert
+                __context__[key] = http_result
+
+    if not __context__.get(key):
+        return []
+
+    alerts = __context__[key].values()
+
+    if metric_name:
+        return __context__[key].get(metric_name)
+
+    return [alert['_id'] for alert in alerts if '_id' in alert]
+
+
+def get_notification_channel(notify_channel, profile="telemetry"):
+    '''
+    Given an email address, creates a notification-channels
+    if one is not found and also returns the corresponsing
+    notification channel id.
+
+    notify_channel
+        Email escalation policy
+    profile
+        A dict of telemetry config information.
+    '''
+    # This helper is used to procure the channel ids
+    # used to notify when the alarm threshold is violated
+
+    auth = _auth(profile=profile)
+
+    notification_channel_id = _retrieve_channel_id(notify_channel)
+
+    if not notification_channel_id:
+        log.info("{0} channel does not exist, creating.".format(notify_channel))
+
+        # create the notification channel and cache the id
+        post_url = _get_telemetry_base(profile) + "/notification-channels"
+        data = {
+            "_type": "EmailNotificationChannel",
+            "name":  notify_channel[:notify_channel.find('@')] + 'EscalationPolicy',
+            "email": notify_channel
+        }
+        response = requests.post(post_url, data=json.dumps(data), headers=auth)
+        if response.status_code == 200:
+            log.info("Successfully created EscalationPolicy {0} with EmailNotificationChannel {1}"
+                .format(data.get('name'), notify_channel))
+            notification_channel_id = response.json().get('_id')
+            __context__["telemetry.channels"][notify_channel] = notification_channel_id
+        else:
+            raise Exception("Failed to created notification channel {0}".format(notify_channel))
+
+    return notification_channel_id
+
+
+def get_alarms(deployment_id, profile="telemetry"):
+    """ get all the alarms set up against the current deployment
```

Comment:
Can you change these (and any other docstrings in this file) to single quotes? We like the `'''` version instead of `"""`.